### ISSUE TITLE
[Bug] TensorRT conversion error

### ISSUE THREAD

**Issue by mike199515** [tensorflow/tensorflow#22577]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:No
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**: 1.11.0 dev build from dockerhub, last commit c19e29306ce1777456b2dbb3a14f511edf7883a8
- **Python version**:2.7
- **Bazel version (if compiling from source)**:0.15.0
- **GCC/Compiler version (if compiling from source)**:5.4.0
- **CUDA/cuDNN version**: 9.0/7.0.5
- **GPU model and memory**: GTX 1080 with 8G memory
- **Exact command to reproduce**: python minimal_graph.py 

### Describe the problem
Met with another conversion error when using latest tensorRT conversion pipeline. After localization of the problem, it seems that the problem is caused by NHWC -> NCHW conversion: If we have a constant of shape (C) (usually bias add term) added after the convolution, after the NHWC -> NCHW conversion, we should expect it to be of shape (C, 1, 1) in order to be broadcast-able, yet current pipeline failed to do so.
My guess may be incorrect, hopefully this would be helpful though.

@aaroey 

### Source code / logs
Test case:
```
 import numpy as np
 import tensorflow as tf
 from tensorflow.contrib import tensorrt as trt
 
 def build_graph_from_def(graph_def, input_nodes, output_nodes):
     """
     build the actual graph from definition
     """
     tf.reset_default_graph()
     graph = tf.Graph()
     with graph.as_default():
         return_tensors = [operation_name + ":0" for operation_name in input_nodes + output_nodes]
         tensors = tf.import_graph_def(graph_def=graph_def, name="",
                                       return_elements=return_tensors)
         input_tensor_list = tensors[:len(input_nodes)]
         output_tensor_list = tensors[len(input_nodes):]
 
     return graph, input_tensor_list, output_tensor_list
 
 
 def main():
 
     with tf.variable_scope("Net"):
         inp = tf.placeholder(tf.float32, shape=(1, 32, 32, 3), name="input_image")
         weight = tf.zeros([3,3,3,8] ,tf.float32)
         conv = tf.nn.conv2d(inp, weight, strides=[1,1,1,1], padding="SAME")
         bn = conv + tf.ones((8,), tf.float32, name="add")
         bn = tf.nn.relu(bn, name="output")
 
     input_nodes = ["Net/input_image"]
     output_nodes = ["Net/output"]
 
     with tf.Session() as sess:
         sess.run(tf.global_variables_initializer())
         const_graph_def = tf.graph_util.convert_variables_to_constants(
             sess, sess.graph.as_graph_def(), output_nodes)
 
     optimized_graph_def = trt.create_inference_graph(
         input_graph_def=const_graph_def,
         outputs=output_nodes,
         max_batch_size=1,
         max_workspace_size_bytes=1 << 25)
     graph, input_tensors, output_tensors = build_graph_from_def(
         optimized_graph_def, input_nodes, output_nodes)
 
     with tf.Session(graph=graph) as sess:
         output_value = sess.run(output_tensors[0], feed_dict={input_tensors[0]: np.zeros((1, 32,    32, 3))})
     print("output:{}".format(output_value.shape))
 
 if __name__ == "__main__":
     main()
```
Sample Output:
```
$ python minimal_graph2.py
2018-09-27 17:44:41.357261: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-09-27 17:44:41.357805: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1411] Found device 0 with properties: 
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.7335
pciBusID: 0000:01:00.0
totalMemory: 7.92GiB freeMemory: 5.67GiB
2018-09-27 17:44:41.357817: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2018-09-27 17:44:41.632033: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-09-27 17:44:41.632053: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2018-09-27 17:44:41.632058: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2018-09-27 17:44:41.632205: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 5419 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-09-27 17:44:41.689349: I tensorflow/core/grappler/devices.cc:51] Number of eligible GPUs (core count >= 8): 1
2018-09-27 17:44:41.689390: I tensorflow/core/grappler/clusters/single_machine.cc:359] Starting new session
2018-09-27 17:44:41.689539: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2018-09-27 17:44:41.689552: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-09-27 17:44:41.689557: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2018-09-27 17:44:41.689562: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2018-09-27 17:44:41.689643: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 5419 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-09-27 17:44:41.693257: I tensorflow/contrib/tensorrt/convert/convert_nodes.cc:2952] Segment @scope 'Net/', converted to graph
2018-09-27 17:44:41.693268: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:418] Can't find a device placement for the op!
python: customWinogradConvActLayer.cpp:48: nvinfer1::cudnn::WinogradConvActLayer::WinogradConvActLayer(const string&, const EngineTensors&, const EngineTensors&, const nvinfer1::ConvolutionParameters&, bool, const std::vector<float>&): Assertion `matchNbDims(inputs[0], outputs[0]) && (inputs.size() == 1 || inputs[1].extent == outputs[0].extent)' failed.
[1]    28090 abort (core dumped)  python minimal_graph2.py

```

**Comment by aaroey** [comment]
@pooyadavoodi

**Comment by abhishek-neurala** [comment]
I've been running into the same issue with an almost similar scenario and exactly same environment. Can we please get some activity going  on for this issue?

**Comment by wt-huang** [comment]
@mike199515 I ran the above code snippet without running into any issues. Try to install TensorFlow from binary. Also, please post your `env.txt` here after running `tf_env_collect.sh`.

**Comment by wt-huang** [comment]
Closing this issue, feel free to reopen if any error comes up.

**Comment by mike199515** [comment]
Hello, I tried running on latest 1.11.0 tensorflow-gpu installed from pip, and I still got the same output as described above.
My env.txt looks like the following:
```

== cat /etc/issue ===============================================
Linux yiwei-desktop 4.15.0-36-generic #39~16.04.1-Ubuntu SMP Tue Sep 25 08:59:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
VERSION="16.04.5 LTS (Xenial Xerus)"
VERSION_ID="16.04"
VERSION_CODENAME=xenial

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


== uname -a =====================================================
Linux yiwei-desktop 4.15.0-36-generic #39~16.04.1-Ubuntu SMP Tue Sep 25 08:59:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy                              1.15.2                
protobuf                           3.6.0                 
tensorflow-gpu                     1.11.0                
tensorflow-tensorboard             0.1.8                 

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.11.0
tf.GIT_VERSION = v1.11.0-0-gc19e29306c
tf.COMPILER_VERSION = v1.11.0-0-gc19e29306c
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH is unset
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
Thu Nov  1 18:42:17 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.130                Driver Version: 384.130                   |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1080    Off  | 00000000:01:00.0  On |                  N/A |
| 29%   46C    P2    42W / 180W |   1391MiB /  8112MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      2190      G   /usr/lib/xorg/Xorg                           582MiB |
|    0      5166      G   compiz                                       493MiB |
|    0      6354      G   ...-token=6EB0A193120E130143DC89992F94A977   196MiB |
|    0      7163      C   /usr/lib/libreoffice/program/soffice.bin     107MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudart.so.9.0.252
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-9.0/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.0/doc/man/man7/libcudart.7
```

Hope it helps.

**Comment by mike199515** [comment]
@wt-huang I switched my model to NCHW format to avoid this problem, yet it would still be helpful if we can fix this? I think I don't have the permission to re-open the issue yet..

**Comment by wt-huang** [comment]
@mike199515 This should work with NCHW format, which is more the norm. We will note down to add NHWC -> NCHW conversion to make it backward compatible.

**Comment by mike199515** [comment]
To be clear, it's not an issue with the NHWC -> NCHW conversion, rather this is a problem with the NHWC format which is the default in tensorflow, so I would expect this to affect the majority of use cases

**Comment by sdlpkxd** [comment]
I encounter with the same problem, is there any official resolution ?

**Comment by smit-hinsu** [comment]
sorry for the delayed response for this issue.

I was able to reproduce this issue with TF 1.12, TensorRT 4 and CUDA 9.0 using the minimal example provided in https://github.com/tensorflow/tensorflow/issues/23467#issue-376975195. But, the good news is that the issue seems to be fixed with nightly TensorFlow (and soon to be released TF 1.13), TensorRT 5 and CUDA 10.0.

Please let us know know if that does not fix the issue for you. Otherwise, will close this issue after waiting for a week.

Also, adding @pooyadavoodi @trevor-m to shed light on what actually fixed the issue if that is already known.

**Comment by sdlpkxd** [comment]
Good news, thanks very much.  : )
